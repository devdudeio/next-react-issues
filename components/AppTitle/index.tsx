import PropTypes from 'prop-types';
import {FunctionComponent} from 'react';


const AppTitle: FunctionComponent = ({children, ...props}): JSX.Element =>
    <h1 className="text-white text3xl font-extrabold" {...props}>{children}</h1>

AppTitle.propTypes = {
    children: PropTypes.node.isRequired,
};

AppTitle.defaultProps = {
    children: "missing children"
}

export default AppTitle;