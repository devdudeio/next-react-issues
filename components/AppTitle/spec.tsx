import renderer from "react-test-renderer";
import AppTitle from "./index";

describe('AppTitle', ()=> {
    test('should render with text', () => {
        const component = renderer.create(
            <AppTitle>React Issues</AppTitle>,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('should render default props', () => {
        const component = renderer.create(
            <AppTitle />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
})
