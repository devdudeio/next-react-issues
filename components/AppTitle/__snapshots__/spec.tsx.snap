// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTitle should render default props 1`] = `
<h1
  className="text-white text3xl font-extrabold"
>
  missing children
</h1>
`;

exports[`AppTitle should render with text 1`] = `
<h1
  className="text-white text3xl font-extrabold"
>
  React Issues
</h1>
`;
